src/Generator/Generators/CLI/CLISources.cs

```diff
@@ -1173,10 +1173,23 @@ public struct ParamMarshal
             var argName = Generator.GeneratedIdentifier("arg") + paramIndex.ToString(CultureInfo.InvariantCulture);
 
             var isRef = param.IsOut || param.IsInOut;
+
+            var paramType = param.Type;
```

Comment:
Please add desugaring at the end, delete the rest of your change in this file and try again.

Reply:
can you elaborate please? At the end where exactly?